### ISSUE TITLE
stratify implementation in holdout pipeline

### ISSUE THREAD

**Issue by everglory99** [instacart/lore#124]
Hi, 
I am a little confused about the stratify implementation in ` _split_data` method in `lore.pipelines.holdout.base`. From my understanding and some experiment, this is different than `stratify` option in `sklearn.model_selection.train_test_split` (see [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html)). Would you please elaborate why you implement `stratify` in this way? Thanks very much!

**Comment by montanalow** [comment]
It might help me understand the question if you provide an example of the confusing behavior. `stratify` is used to place all of the a single categorical's data in a single set (train/validate/test) to prevent leakage. 

e.g. `stratify = 'user_id'` would ensure that all the data for a single user must be kept together, to prevent a habituated user from leaking signal between train and test.

**Comment by everglory99** [comment]
Thanks for the explanation. The implementation does make sense. I guess we just have different perspectives on "stratify". In `sklearn.model_selection.train_test_split`, `stratify` is to return training and test subsets that have the same proportions of class labels as the input dataset. 

For example, if label `y` is a binary variable with values 0 and 1 and there are 15% of zeros and 85% of ones, `stratify=y` will make sure that both train and test set have 15% of 0's and 85% of 1's. This is similar to [`sklearn.model_selection.StratifiedKFold`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html)

**Comment by montanalow** [comment]
@everglory99 Thanks for clarifying. I'm going to close. Please feel free to reopen if you have further thoughts.
